src/item_factory.cpp

```diff
@@ -1932,15 +1932,32 @@ bool load_min_max(std::pair<T, T> &pa, JsonObject &obj, const std::string &name)
     return result;
 }
 
-bool load_sub_ref(std::unique_ptr<Item_spawn_data> &ptr, JsonObject &obj, const std::string &name)
+bool Item_factory::load_sub_ref( std::unique_ptr<Item_spawn_data> &ptr, JsonObject &obj,
+                                 const std::string &name, const Item_group &parent )
 {
     if (obj.has_member(name)) {
         // TODO!
-    } else if (obj.has_member(name + "-item")) {
+    } else if( obj.has_array( name + "-item" ) || obj.has_array( name + "-group" ) ) {
+        if( name != "contents" ) {
+            debugmsg( "It doesn't make sense to use an array for %s (see doc/ITEM_SPAWN.md)", name.c_str() );
+            return false;
+        }
+        const bool isgroup = obj.has_array( name + "-group" );
```

Comment:
The flag means the function prefers `-group` to `-item`. If both were specified, the latter would be silently ignored. This can be quite confusing for those who would use them. Does it make sense to prohibit specifying `-group` and `-item` together (for the single strings too)?